build/api-plugin.js

```diff
@@ -0,0 +1,163 @@
+} */
+
+const capitalize = str => {
+  return str.charAt(0).toUpperCase() + str.slice(1)
+}
+
+/* const hyphenateRE = /\B([A-Z])/g
+function hyphenate (str) {
+  return str.replace(hyphenateRE, '-$1').toLowerCase()
+} */
+
+function genFrontMatter (component) {
+  const fm = [
+    `title: ${component} API`,
+    `description: API for the ${component} component.`,
+    `keywords: ${component}, api, vuetify`,
+  ]
+  return `---\n${fm.join('\n')}\n---\n\n`
+}
+
+function genTableHeader (headers) {
+  const headerLine = []
+  const dividerLine = []
+  for (const header of headers) {
+    headerLine.push(`${capitalize(header)}`)
+    dividerLine.push('---')
+  }
+  return [
+    genRowString(headerLine),
+    genRowString(dividerLine),
+  ]
+}
+
+function genTableRow (headers, row) {
+  const headerRow = []
+  for (const header of headers) {
+    if (header === 'source' && isProduction) continue
+
+    let value = row[header]
+
+    if (['default', 'value', 'signature'].includes(header)) {
+      value = `\`${row[header]}\``
+    } else if (header === 'name') {
+      value = `<div class="font-weight-bold text-mono">${row[header]}</div>`
+    } else if (header === 'type') {
+      value = `<div class="text-mono">${row[header]}</div>`
+    }
+
+    headerRow.push(value || '')
+  }
+  return genRowString(headerRow)
+}
+
+function genRowString (row) {
+  return `| ${row.join(' | ')} |`
+}
+
+function genTable (tableData) {
+  const headers = Object.keys(tableData[0])
+  const table = genTableHeader(headers)
+
+  for (const row of tableData) {
+    table.push(genTableRow(headers, row))
+  }
+
+  return `${table.join('\n')}\n\n`
+}
+
+function genFooter () {
+  const footer = [
+    '<carbon-ad />',
+    '<up-next />',
+    '<vuetify-ad />',
+    '<contribute />',
+  ]
+  return `${footer.join('\n\n')}\n\n`
+}
+
+function writeMdFile (file, data) {
+  const root = './src/pages/en/api'
+  const stream = createWriteStream(`${root}/${file}.md`)
+
+  // gen front-matter
+  const frontMatter = genFrontMatter()
+  const title = '# V-Alert API\n\n'
+  const footer = genFooter()
+
+  stream.once('open', () => {
+    stream.write(frontMatter)
+    stream.write(title)
+    for (const [header, value] of Object.entries(data)) {
+      if (header !== 'mixins') {
+        const headerLine = header === 'sass'
+          ? '## SASS Variables\n'
+          : `## ${capitalize(header)}\n`
+        const table = genTable(value)
+        stream.write(headerLine)
+        stream.write(table)
+      }
+    }
+    stream.write(footer)
+    // stream.write(JSON.stringify(obj, null, 2))
+    stream.end()
+  })
+}
+
+function createMdFile (component, data) {
+  let str = ''
+
+  str += genFrontMatter(component)
+  str += `# ${component} API\n\n`
+  for (const [header, value] of Object.entries(data)) {
+    if (header === 'mixins') continue
+
+    const headerLine = header === 'sass'
+      ? '## SASS Variables\n'
```

Comment:
Same here, does this need to be translated?

Reply:
That's something I thought about as well while doing this. But I chose not to for two reasons.

1) Current docs does _not_ translate headers related in API section such as "Props", "Slots", "Name", "Description", etc
2) Not doing it is easier

It would definitely be possible if we want to. I figure we could import the i18n plugin and use `i18n.t()`.

Reply:
I think for now, as long as this is the behavior for all of the generated sections in the api, we can revisit it later.